GameServerLib/Scripting/CSharp/CSharpScriptEngine.cs

```diff
@@ -53,50 +72,58 @@ public bool Load(List<string> scriptLocations)
             var assemblyName = Path.GetRandomFileName();
 
             var references = new List<MetadataReference>();
-            foreach (var a in AppDomain.CurrentDomain.GetAssemblies())
-                if (!a.IsDynamic && !a.Location.Equals(""))
-                    references.Add(MetadataReference.CreateFromFile(a.Location));
+            foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies())
+            {
+                if (!assembly.IsDynamic && !assembly.Location.Equals(""))
+                {
+                    references.Add(MetadataReference.CreateFromFile(assembly.Location));
+                }
+            }
+            //Add vector2 reference because vector2 isn't in System.Numerics in dot net core
+            references.Add(MetadataReference.CreateFromFile(typeof(Vector2).Assembly.Location));
             //Now add game reference
             references.Add(MetadataReference.CreateFromFile(typeof(Game).Assembly.Location));
-            var op = new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary)
+            var compilationOptions = new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary)
                 .WithOptimizationLevel(OptimizationLevel.Release).WithConcurrentBuild(true);
 
             var compilation = CSharpCompilation.Create(
                 assemblyName,
                 treeList,
                 references,
-                op
+                compilationOptions
             );
 
-            var errored = false;
+            var state = CompilationStatus.Compiled;
             while (true)
-                using (var ms = new MemoryStream())
+            {
+                using (var compiledAssemblies = new MemoryStream())
                 {
-                    var result = compilation.Emit(ms);
+                    var result = compilation.Emit(compiledAssemblies);
 
                     if (result.Success)
                     {
-                        ms.Seek(0, SeekOrigin.Begin);
-                        var assembly = Assembly.Load(ms.ToArray());
+                        compiledAssemblies.Seek(0, SeekOrigin.Begin);
+                        var assembly = Assembly.Load(compiledAssemblies.ToArray());
                         _scriptAssembly.Add(assembly);
                         foreach (var type in assembly.GetTypes())
                         {
                             types.Add(type.FullName, type);
                         }
+                        return state;
                     }
-
-                    errored |= true;
-                    var invalidSourceTrees = GetInvalidSourceTrees(result);
-
-                    if (invalidSourceTrees.Count == 0)
+                    else
                     {
-                        // Shouldnt happen
-                        _logger.Error("Script compilation failed");
-                        return true;
+                        state = CompilationStatus.SomeCompiled;
```

Comment:
Does nothing?

Reply:
That changes the state from everything compiled to only some things compiled. This is enum is checked outside this class to print out a message stating only some of the scripts compiled.
You can test that this works by introducing an error in one script and seeing the console output that only some of the scripts compiled.

Reply:
It gets returned in the next while loop run if the compilation is successful.